### ISSUE TITLE
Connection dropping regularly since AT4 update a couple of days ago

### ISSUE THREAD

**Issue by smeagol-github** [mihailescu2m/hass-airtouch4-platform#11]
Good afternoon!

I'm having a lot of issues with the connection between HA and AT4 dropping out until I make a change to a temperature or mode. 

After I make that change I see this in the logs:
```
2023-12-02 16:26:12.448 ERROR (MainThread) [custom_components.polyaire.airtouch4] Connection error in receiver!
2023-12-02 16:26:12.448 INFO (MainThread) [custom_components.polyaire.airtouch4] Message receiver lost connection, trying to reconnect...
2023-12-02 16:26:12.448 INFO (MainThread) [custom_components.polyaire.airtouch4] (Re)connecting...
2023-12-02 16:26:12.448 DEBUG (MainThread) [custom_components.polyaire.airtouch4] open socket connection to the airtouch...
2023-12-02 16:26:12.451 INFO (MainThread) [custom_components.polyaire.airtouch4] (Re)connected!
2023-12-02 16:26:12.451 INFO (MainThread) [custom_components.polyaire.airtouch4] Message receiver task (re)started...
2023-12-02 16:27:07.932 DEBUG (MainThread) [custom_components.polyaire.airtouch4] Message received with expected header!
2023-12-02 16:27:07.932 DEBUG (MainThread) [custom_components.polyaire.airtouch4] Message received is group message!
```
Followed by the usual messages about updating entities. 

I know Wifi isn't the problem because it's turned off on the tablet and I'm using an ethernet dongle - which I ping test independently.

Current Console Version: 1.4.4
Current Main Module Version: 2.1.0.2

Please let me know if there is any additional information required to diagnose.

**Comment by Skrae86** [comment]
Mine is borked as well, log is full of connection fail errors. I've called Polyaire and will try to sus if i can roll back firmware as 1.4.3 was working great

**Comment by smeagol-github** [comment]
@Skrae86 let me know how you go - I haven't had a chance to call them yet.

**Comment by spry-salt** [comment]
Mine is doing the same. I’ve got a newer tablet but an AT4 system. The newer tablet lets it connect via 5GHz wifi but that seemed to be flaky these last couple of weeks so I changed the network back to just 2.4GHz. Now I notice that when the tablet drops off the wifi connection, it needs a reboot to be able to even recognise that ANY wifi networks are out there. Never seen this before in almost 2 years of use.

**Comment by spry-salt** [comment]
I get the same kind of frequent connection errors whether I use the official AirTouch integration or the other one from HACS.

**Comment by spry-salt** [comment]
The connection drops regularly whether I'm using the default AirTouch integration or this one.

**Comment by Gibleh** [comment]
> @Skrae86 let me know how you go - I haven't had a chance to call them yet.

@spry-salt @smeagol-github @Skrae86 @mihailescu2m 

TLDR: I found a way to downgrade and thought I'd share.

Hey team, I had this same problem with version 1.4.4, I have found a way to downgrade back to 1.4.3 on the Airtouch 4 wall tablet. It should work for anyone that has had 1.4.3 installed previously.

NOTE: Do at your own risk. It worked for me with no issues. I do not work for Polyaire, nor have I had any contact with them on this topic.

First confirm that you do have the 1.4.3 or older apk on your wall tablet,
Close Airtouch and swipe up from the bottom to open the app draw, select "Files"
Navigate to "Internal Storage" -> "airtouch"
Confirm you have older version apk's available. (As my install is only about 18 months old I only have 1.4.2, 1.4.3 and 1.4.4 available)
OPTIONAL: Insert a micro SD card and copy the apk's to a safe location for future use. You never know when they may integrate a method to remove these files on update.
NOTE: although these apk's have the option to install, they will fail as the currently installed version is newer.

Now that you have confirmed you have the apk/s required for downgrade its time to roll back the existing installation.
Navigate to "Settings" -> "Apps & notifications" -> "AirTouch".
If AirTouch is not listed in the Recently opened apps in the upper part of the screen, select "SEE ALL xx APPS" (where xx is the number of apps on your system) -> "AirTouch".
When in the AirTouch App info screen, select "FORCE STOP". Once the app is stopped, tap the 3-dot menu in the top right corner and select "Uninstall updates".
Your AirTouch tablet will now have the stock, from-factory AirTouch application.

NOTE: If you open the AirTouch app at this point and select the update option, it will install back to 1.4.4. We don't want this, so continue reading below.

Now that we have rolled back to a version earlier than the apk's we checked in at the start, we can install on of these older apk's.

Navigate back to the apk's we located earlier ("Files" -> Internal storage -> airtouch)
Tap the apk of the version you wish to install (likely 1.4.3 as this version works well with the integration this issue is raised on)
When prompted tap Install.

Congrats, you should have successfully downgraded to pre 1.4.4, check the AirTouch app for confirmation (in the AirTouch app select the 3-dot/line menu top right -> System Settings -> Preferences (along the top) -> VERSIONS (button bottom right).
Current Console Version should show the version you chose earlier. There will also likely be a prompt for updating. Yeah, nah, don't do that.

Apologies for the wall of text, but I hope this helps someone.

**Comment by Skrae86** [comment]
Thanks for that @Gibleh I was poking around yesterday and saw that I had the Apks from older versions and was going to look at how to do it this weekend. Thanks for the writeup I'll be doing that asap.

I've had SFA from Polyaire and they have ghosted further interactions on the matter (messaging the tech directly on his mobile)

will rollback and see how things go =)

**Comment by spry-salt** [comment]
Some new information on recent connection issues is contained in this post on Whirlpool but I don’t know enough to fix it. If correct it seems to me it would affect both the AirTouch integrations that are available:

`https://forums.whirlpool.net.au/thread/3qqplnz3?p=16#r73690731`

**Comment by dan-dan81** [comment]
Anyone had any luck here? I've downgraded to 1.4.3 but still have similar issues.

**Comment by smeagol-github** [comment]
Mate I actually found mine was fixed by completely deactivating IPv6 on the router for that network segment. 

Fixed it instantly. 

Which I think I found a reference to on whirlpool somewhere.
